terraform/mgmtcluster.tf

```diff
@@ -54,14 +65,18 @@ resource "openstack_compute_instance_v2" "mgmtcluster_server" {
 
   network { port = openstack_networking_port_v2.mgmtcluster_port.id }
   block_device {
-    uuid                  = data.openstack_images_image_ids_v2.images.ids[0]
-    source_type           = "image"
-    volume_size           = 30
+    uuid                  = openstack_blockstorage_volume_v3.mgmtcluster_volume.id
+    source_type           = "volume"
     boot_index            = 0
     destination_type      = "volume"
     delete_on_termination = true
   }
 
+  lifecycle {
```

Comment:
I had to read https://itnext.io/how-and-when-to-ignore-lifecycle-changes-in-terraform-ed5bfb46e7ae to understand why this makes sense ...

Reply:
😁 Like it does not need to be there.

I originally was trying to accomplish that the volume would get created only when running on a new instalation, but that was not possible as that was cycle of dependencies.

Terraform apply is run only in make create. So unless someone runs make create again or runs apply manually this should not be needed.

If they did run it without this the management node root volume would get destroyed along with all data. 

With this it will create a new volume however the old one will stay attached and existing.

So it is backwards compatible and relatively safe except of possible creation of a new unused volume when user does something they should not.